### PR TITLE
Add https and http proxy support for fetch

### PR DESCRIPTION
**Pull request by yageek**
Let swift build be aware of the http proxy variables.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
I wonder if we should inherit all of the env

**Comment by yageek** [comment]
I did it this way to be consistent with the other function that inject the `environment` variable into the `system` function. 

Do you think it would be better to split the variable list passed ?

**Comment by ddunbar** [comment]
I am coming around to thinking that we should inject the full environment when we are running git clone commands, rather than cherry pick variables. That should solve the problem here, can we do that instead?

**Comment by yageek** [comment]
As suggested, I updated the PR to read all the environment variable for the cloning operations.

File: Sources/Utility/Git.swift [hunk 1]
```diff
@@ -129,25 +130,11 @@ public class Git {
...
-            "TERM",
-            "XDG_CONFIG_HOME",
-        ]
-        var result = [String: String]()
-        for name in environmentList {
-            result[name] = getenv(name)
-        }
-        return result
-    }()
+    private class func readEnvironmentVariables(names: [String]) -> [String: String] {
```
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: This is probably not needed now right?

File: Sources/Utility/Git.swift [hunk 2]
```diff
@@ -47,11 +48,11 @@ public class Git {
         public var sha: String! {
             return try? Git.runPopen([Git.tool, "-C", path, "rev-parse", "--verify", "HEAD"]).chomp()
         }
-        
+
         public func versionSha(tag: String) throws -> String {
             return try Git.runPopen([Git.tool, "-C", path, "rev-parse", "--verify", "\(tag)"]).chomp()
         }
-        
+
```
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: can you remove these diffs

File: Sources/Utility/Git.swift [hunk 3]
```diff
@@ -47,11 +48,9 @@ public class Git {
         public var sha: String! {
             return try? Git.runPopen([Git.tool, "-C", path, "rev-parse", "--verify", "HEAD"]).chomp()
         }
-        
         public func versionSha(tag: String) throws -> String {
             return try Git.runPopen([Git.tool, "-C", path, "rev-parse", "--verify", "\(tag)"]).chomp()
         }
-
```
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: Can you discard these diffs? If you're having trouble looking at whitespaces diff you can use a good GUI client which can show the whitespace diff properly. (I sometimes use gitX however its quite old... I am sure very good GUI clients are available)

**Comment by ddunbar** [comment]
LGTM

**Comment by ddunbar** [comment]
@swift-ci please test

File: Sources/Utility/Git.swift (hunk 3 above)
Comment (yageek) [inline Sources/Utility/Git.swift]: I will change it

Comment (ddunbar) [inline Sources/Utility/Git.swift]: It's ok, I went ahead and pulled it in.

**Pull request merged**